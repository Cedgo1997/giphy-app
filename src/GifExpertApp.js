import React, { useState } from 'react';
import { Fragment } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import PropTypes from 'prop-types';

export const GifExpertApp = ({ defaultCategories = [] }) => {


    const [categories, setCategories] = useState(defaultCategories);

    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map((category) =>
                        <GifGrid
                            category={category}
                            key={category}
                        />)
                }
            </ol>
        </Fragment>
    )
}

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.arrayOf(String).isRequired
}