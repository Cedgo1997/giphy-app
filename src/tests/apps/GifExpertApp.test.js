import { GifExpertApp } from "../../GifExpertApp";
import React from 'react';
import { shallow } from "enzyme";

describe('Pruebas en GifExpertApp', () => {
    const defaultCategories = ['Dragon Ball', 'Naruto', 'Saint Seiya'];
    const wrapper = shallow(<GifExpertApp defaultCategories={defaultCategories} />);
    test('Debe renderizar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe existir el componente AddCategory', () => {
        const addCategory = wrapper.find('AddCategory');
        expect(addCategory.exists()).toBeTruthy();
    });
    test('Debe existir el componente GifGrid', () => {
        const gifGrid = wrapper.find('GifGrid');
        expect(gifGrid.exists()).toBeTruthy();
        expect(gifGrid.length).toBe(defaultCategories.length);
    });
});