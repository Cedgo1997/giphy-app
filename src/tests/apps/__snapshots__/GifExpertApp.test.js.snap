// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe renderizar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Saint Seiya"
      key="Saint Seiya"
    />
  </ol>
</Fragment>
`;
